python/createCollectionDownload.py

```diff
@@ -2,75 +2,189 @@
+    """
+    r = requests.request(url=url, method=method, headers=headers, params=params, data=data)
+    if r.status_code == 401:
+        raise ValueError("Bad ChainXY API key provided, please double-check the input value!")
+    elif r.status_code != 200 and 'does not correspond to a collection' in r.text.lower():
+        print("The collection does not exist in the database.")
+        r = None
+    else:
+        return r
+
+def validate_inputs(headers:str, cxy_api_key:str, collection_type:str, cache_time:int):
+    """
+    Verifies that a given API key is valid for making requests to the API and checks that the collection download
+    inputs are valid for processing.
+    """
     url = 'https://location.chainxy.com/api/Users/Me'
-    headers = {'x-apikey': cxy_api_key,
-            'x-Application': 'Python API Call',
-            'content-type': 'application/json'}
-    response = requests.get(url, headers=headers)
-    if response.status_code == 401:
-        raise ValueError("Bad ChainXY API key provided, double-check the provided value!")
+    r = request_api(url=url, headers=headers)
+    if not isinstance(collection_type, str) or collection_type.lower() not in ['chain', 'center']:
+        raise ValueError(f"Collection type '{collection_type}' must be one of ['chain', 'center'].")
+    
+    if cache_time < 0:
+        raise ValueError("Please enter a valid duration for cache_time that is an int > 0.")
+
+def get_collection_urls(collection_id:str, collection_type:str):
+    """
+    Returns URLs and url params for requesting the ChainXY API for collection records based on collection_type.
+    """
+    records_url = check_url = new_download_url = ''
+
+    if collection_type == 'chain':
+        
+        records_url = f'https://location.chainxy.com/api/Downloads?fields=Id,ChainList,Label,User,ReportType,CreateDate,DataDate,Status,Format,Count,Link,DisplayValue,&Query=%7B%22ChainListId%22:{collection_id}%7D&OrderBy=-CreateDate'
+        check_url = f'https://location.chainxy.com/api/ChainLists/{collection_id}'
+        new_download_url = f"https://location.chainxy.com/api/ChainLists/Download/{collection_id}"
 
+        url_params = {
+            "format": "CSV",  # ZIP_CSV also works
+            "splitLayers": "false",
+            # "dataDate": "2019-10-03" # OPTIONAL
+        }
 
-def download_collection(cxy_api_key:str, collection_id:int):
+    elif collection_type == 'center':
+        
+        records_url = f'https://location.chainxy.com/api/Downloads?fields=Id,SiteList,Label,User,ReportType,CreateDate,DataDate,Status,Format,Count,Link,DisplayValue,&Query=%7B%22SiteListId%22:{collection_id}%7D&OrderBy=-CreateDate'
+        check_url = f'https://location.chainxy.com/api/SiteLists/{collection_id}'
+        new_download_url =  f'https://location.chainxy.com/api/SiteLists/Download/{collection_id}'
+
+        url_params = {
+            "format": "ZIP_CSV"
+        }
+    
```

Comment:
should there be another else here that raises ValueError if the collection type doesn't match the accepted collection_types?

Reply:
The check is in `validate_inputs()`:

```
if not isinstance(collection_type, str) or collection_type.lower() not in ['chain', 'center']:
    raise ValueError(f"Collection type '{collection_type}' must be one of ['chain', 'center'].")
```